C#Backend/InpatientTherapySchedulingProgram/Controllers/HoursWorkedController.cs

```diff
@@ -6,39 +6,49 @@
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.EntityFrameworkCore;
 using InpatientTherapySchedulingProgram.Models;
+using InpatientTherapySchedulingProgram.Services.Interfaces;
+using InpatientTherapySchedulingProgram.Exceptions.HoursWorkedExceptions;
+using InpatientTherapySchedulingProgram.Exceptions.UserExceptions;
 
 namespace InpatientTherapySchedulingProgram.Controllers
 {
     [Route("api/[controller]")]
     [ApiController]
     public class HoursWorkedController : ControllerBase
     {
-        private readonly CoreDbContext _context;
+        private readonly IHoursWorkedService _service;
 
-        public HoursWorkedController(CoreDbContext context)
+        public HoursWorkedController(IHoursWorkedService service)
         {
-            _context = context;
+            _service = service;
         }
 
-        // GET: api/HoursWorkeds
-        [HttpGet]
-        public async Task<ActionResult<IEnumerable<HoursWorked>>> GetHoursWorked()
+        // GET: api/HoursWorkeds/5
+        [HttpGet("{id}")]
+        public async Task<ActionResult<HoursWorked>> GetHoursWorked(int hoursWorkedId)
         {
-            return await _context.HoursWorked.ToListAsync();
+            var hoursWorked = await _service.GetHoursWorkedById(hoursWorkedId);
+
+            if (hoursWorked == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(hoursWorked);
         }
 
         // GET: api/HoursWorkeds/5
         [HttpGet("{id}")]
-        public async Task<ActionResult<HoursWorked>> GetHoursWorked(int id)
+        public async Task<ActionResult<HoursWorked>> GetHoursWorkedByUserId(int hoursWorkedId)
```

Comment:
I am just wondering how this class works from a usage standpoint. Does the frontend just make a request for an hours worked for a user and just gets one back? I think it would make a lot of sense to create 2 hours worked end points. Similar to the therapist event where get requests are also passed in an HoursWorked JSON object. One endpoint would gather all hours worked between the start and end date. Another would do the same thing but would also match based on the UserId/TherapistId passed in. See therapistEvent controller/service for more details.

Reply:
The way the class works is you pass in a userId and get all their hours worked, or you can pass in the specific ID that goes with that exact log to pull that individual hours worked.

Reply:
Ok does this actually return a list of hours worked? It looks like it returns only once hoursWorked. Doesn't it need to be IEnumerable to be a list of objects?